### PR TITLE
missing header inttypes.h

### PR DESCRIPTION
**Pull request by Rehan-MALAK**
most of the platforms have inttypes.h via the header contiki-conf.h 

but sky and simplelink do not include it by default

this PR adds inttypes.h when needed for these latter platforms...

### CODE REVIEW

**Comment by atiselsts** [comment]
I wondering if there is any reason to avoid including it in `contiki-conf.h` on these platforms?

**Comment by Rehan-MALAK** [comment]
I don't know... I push another branch PR #669 in the case we prefer <inttypes.h> included in contiki-conf.h for platform sky and simplelink

**Comment by g-oikonomou** [comment]
I am in favour of including inttypes in files that need inttypes, instead of using a big sledgehammer called contiki-conf.h to include things that other files forget to include :)

File: examples/multicast/root.c [hunk 1]
```diff
@@ -45,6 +45,7 @@
 #include "net/ipv6/multicast/uip-mcast6.h"
 
 #include <string.h>
+#include <inttypes.h>
```
Comment (g-oikonomou) [inline examples/multicast/root.c]: Why do we need this here? Is there a compile failure if we don't include this?

**Comment by Rehan-MALAK** [comment]
example/multicast/root.c is using the the PRIx32 macro : 

```
$ make TARGET=simplelink BOARD=sensortag/cc1350 root.o
  CC        root.c
In file included from ../../os/net/ipv6/uip-debug.h:45:0,
                 from root.c:50:
root.c: In function 'multicast_send':
root.c:86:23: error: expected ')' before 'PRIx32'
   PRINTF(" (msg=0x%08"PRIx32")", uip_ntohl(*((uint32_t *)buf)));
                       ^
../../os/net/net-debug.h:66:28: note: in definition of macro 'PRINTF'
 #define PRINTF(...) printf(__VA_ARGS__)
                            ^~~~~~~~~~~
root.c:86:10: error: conversion lacks type at end of format [-Werror=format=]
   PRINTF(" (msg=0x%08"PRIx32")", uip_ntohl(*((uint32_t *)buf)));
          ^
../../os/net/net-debug.h:66:28: note: in definition of macro 'PRINTF'
 #define PRINTF(...) printf(__VA_ARGS__)
                            ^~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [../../Makefile.include:355: root.o] Error 1
```

**Pull request merged**